lib/puppet/provider/datacat_collector/datacat_collector.rb

```diff
@@ -19,6 +19,9 @@ def exists?
       data.merge!(fragment[:data], &deep_merge)
     end
 
+    # we don't want sensistive data, like AWS access/secret keys, to be populated in puppet debug logs so we delete them here
+    keys_to_delete = ["aws_access_key_id", "aws_secret_access_key"]
+    data.slice!(keys_to_delete)
```

Comment:
I would rather have a flag to not print the collected data instead of special casing key id.

Reply:
Ok, I changed it to a parameter passed into data that lets a user specify what keys to exclude from debug